Source/ablastr/fields/MultiFabRegister.cpp

```diff
@@ -0,0 +1,669 @@
+        const std::string& name,
+        int finest_level
+    )
+    {
+        MultiLevelVectorField field_on_level;
+        field_on_level.reserve(finest_level+1);
+
+        // TODO: Technically, we should search field_on_level via std::unique_copy
+        const std::vector<Direction> all_dirs = {Direction{0}, Direction{1}, Direction{2}};
+
+        for (int lvl = 0; lvl <= finest_level; lvl++)
+        {
+            // insert a new level
+            field_on_level.push_back(VectorField{});
+
+            // insert components
+            for (const Direction& dir : all_dirs)
+            {
+                field_on_level[lvl][dir] = get(name, dir, lvl);
+            }
+        }
+        return field_on_level;
+    }
+
+    ConstMultiLevelVectorField
+    MultiFabRegister::get_mr_levels_alldirs  (
+        const std::string& name,
+        int finest_level
+    ) const
+    {
+        ConstMultiLevelVectorField field_on_level;
+        field_on_level.reserve(finest_level+1);
+
+        // TODO: Technically, we should search field_on_level via std::unique_copy
+        const std::vector<Direction> all_dirs = {Direction{0}, Direction{1}, Direction{2}};
+
+        for (int lvl = 0; lvl <= finest_level; lvl++)
+        {
+            // insert a new level
+            field_on_level.push_back(ConstVectorField{});
+
+            // insert components
+            for (const Direction& dir : all_dirs)
+            {
+                field_on_level[lvl][dir] = get(name, dir, lvl);
+            }
+        }
+        return field_on_level;
+    }
+
+    std::vector<std::string>
+    MultiFabRegister::list () const
+    {
+        std::vector<std::string> names;
+        names.reserve(m_mf_register.size());
+        for (auto const & str : m_mf_register) { names.push_back(str.first); }
+
+        return names;
+    }
+
+    void
+    MultiFabRegister::erase (
+        std::string name,
+        int level
+    )
+    {
+        name = mf_name(name, level);
+
+        if (m_mf_register.count(name) != 1) {
+            throw std::runtime_error("MultiFabRegister::erase name does not exist in register: " + name);
+        }
+        m_mf_register.erase(name);
+    }
+
+    void
+    MultiFabRegister::erase (
+        std::string name,
+        Direction dir,
+        int level
+    )
+    {
+        name = mf_name(name, dir, level);
+
+        if (m_mf_register.count(name) != 1) {
+            throw std::runtime_error("MultiFabRegister::erase name does not exist in register: " + name);
+        }
+        m_mf_register.erase(name);
+    }
+
+    void
+    MultiFabRegister::clear_level (
+        int level
+    )
+    {
+        // C++20: Replace with std::erase_if
+        for (auto first = m_mf_register.begin(), last = m_mf_register.end(); first != last;)
+        {
+            if (first->second.m_level == level)
+                first = m_mf_register.erase(first);
+            else
+                ++first;
```

Comment:
Clang-tidy error "statement should be inside braces [readability-braces-around-statements,-warnings-as-errors]":
```suggestion
            if (first->second.m_level == level) {
                first = m_mf_register.erase(first);
            } else {
                ++first;
            }
```